iceoryx2-bb/posix/src/ipc_capable.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{
+    cell::UnsafeCell,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+pub(crate) mod internal {
+    use std::fmt::Debug;
+
+    use super::*;
+
+    #[derive(Debug, Clone, Copy, Eq, PartialEq)]
+    pub enum Capability {
+        InterProcess,
+        ProcessLocal,
+    }
+
+    /// Stores an OS handle to some resource that is also inter-process capable, like a unnamed
+    /// semaphore or mutex handle.
+    pub struct HandleStorage<T> {
+        handle: UnsafeCell<T>,
+        is_inter_process_capable: AtomicBool,
+        state: AtomicBool,
```

Comment:
What does state mean in this context? `true`/`false` is not quite descriptive. What do you think of another name e.g. `active`, `initialized` or similar?